doc/man7/ossl-guide-quic-client-non-block.pod

```diff
@@ -0,0 +1,453 @@
+such that it might succeed on the retry attempt, but this does not have to be
+the case. It can retry at any time.
+
+Note that it is important that you retry exactly the same operation that you
+tried last time. You cannot start something new. For example if you were
+attempting to write the text "Hello World" and the operation failed because the
+B<SSL> object is currently unable to write, then you cannot then attempt to
+write some other text when you retry the operation.
+
+In this demo application we will create a helper function which simulates doing
+other work. In fact, for the sake of simplicity, it will do nothing except wait
+for the state of the underlying socket to change or until a timeout expires
+after which the state of the B<SSL> object might have changed. We will call our
+function C<wait_for_activity()>.
+
+    static void wait_for_activity(SSL *ssl)
+    {
+        fd_set wfds, rfds;
+        int width, sock, isinfinite;
+        struct timeval tv;
+        struct timeval *tvp = NULL;
+
+        /* Get hold of the underlying file descriptor for the socket */
+        sock = SSL_get_fd(ssl);
+
+        FD_ZERO(&wfds);
+        FD_ZERO(&rfds);
+
+        /*
+         * Find out if we would like to write to the socket, or read from it (or
+         * both)
+         */
+        if (SSL_net_write_desired(ssl))
+            FD_SET(sock, &wfds);
+        if (SSL_net_read_desired(ssl))
+            FD_SET(sock, &rfds);
+        width = sock + 1;
+
+        /*
+         * Find out when OpenSSL would next like to be called, regardless of
+         * whether the state of the underlying socket has changed or not.
+         */
+        if (SSL_get_event_timeout(ssl, &tv, &isinfinite) && !isinfinite)
+            tvp = &tv;
+
+        /*
+         * Wait until the socket is writeable or readable. We use select here
+         * for the sake of simplicity and portability, but you could equally use
+         * poll/epoll or similar functions
+         *
+         * NOTE: For the purposes of this demonstration code this effectively
+         * makes this demo block until it has something more useful to do. In a
+         * real application you probably want to go and do other work here (e.g.
+         * update a GUI, or service other connections).
+         *
+         * Let's say for example that you want to update the progress counter on
+         * a GUI every 100ms. One way to do that would be to use the timeout in
+         * the last parameter to "select" below. If the tvp value is greater
+         * than 100ms then use 100ms instead. Then, when select returns, you
+         * check if it did so because of activity on the file descriptors or
+         * because of the timeout. If the 100ms GUI timeout has expired but the
+         * tvp timeout has not then go and update the GUI and then restart the
+         * "select" (with updated timeouts).
+         */
+
+        select(width, &rfds, &wfds, NULL, tvp);
```

Comment:
There is one possibility that could be done here that might be noted - if the timeout happens we could just call SSL_handle_events() in a loop instead of retrying the operation.

Reply:
Isn't this covered by the text "The L<SSL_get_event_timeout(3)> function can be used to determine what the deadline is for the next time we need to call an I/O function (or call L<SSL_handle_events(3)>)."

Reply:
It might not be so clear from the text. But I am OK with the demo & text as is.